cmd/collectors/rest/rest.go

```diff
@@ -494,6 +492,121 @@ func (r *Rest) GetRestData(prop *prop, client *rest.Client) ([]interface{}, erro
 	return records, nil
 }
 
+func (r *Rest) CollectAutoSupport(p *collector.Payload) {
+	var exporterTypes []string
+	for _, exporter := range r.Exporters {
+		exporterTypes = append(exporterTypes, exporter.GetClass())
+	}
+
+	var counters = make([]string, 0)
+	for k := range r.prop.counters {
+		counters = append(counters, k)
+	}
+
+	var schedules = make([]collector.Schedule, 0)
+	tasks := r.Params.GetChildS("schedule")
+	if tasks != nil && len(tasks.GetChildren()) > 0 {
+		for _, task := range tasks.GetChildren() {
+			schedules = append(schedules, collector.Schedule{
+				Name:     task.GetNameS(),
+				Schedule: task.GetContentS(),
+			})
+		}
+	}
+
+	// Add collector information
+	p.AddCollectorAsup(collector.AsupCollector{
+		Name:      r.Name,
+		Query:     r.prop.query,
+		Exporters: exporterTypes,
+		Counters: collector.Counters{
+			Count: len(counters),
+			List:  counters,
+		},
+		Schedules:     schedules,
+		ClientTimeout: r.client.Timeout.String(),
+	})
+
+	if r.Name == "Rest" && (r.Object == "Volume" || r.Object == "Node") {
+		version := r.client.Cluster().Version
+		p.Target.Version = strconv.Itoa(version[0]) + "." + strconv.Itoa(version[1]) + "." + strconv.Itoa(version[2])
+		p.Target.Model = "cdot"
+		p.Target.Serial = r.client.Cluster().Uuid
+		p.Target.ClusterUuid = r.client.Cluster().Uuid
+
+		md := r.GetMetadata()
+		info := collector.InstanceInfo{
+			Count:      md.LazyValueInt64("count", "data"),
+			DataPoints: md.LazyValueInt64("datapoint_count", "data"),
+			PollTime:   md.LazyValueInt64("poll_time", "data"),
+			ApiTime:    md.LazyValueInt64("api_time", "data"),
+			ParseTime:  md.LazyValueInt64("parse_time", "data"),
+			PluginTime: md.LazyValueInt64("plugin_time", "data"),
+		}
+
+		if r.Object == "Node" {
+			nodeIds, err := r.getNodeUuids()
+			if err != nil {
+				// log but don't return so the other info below is collected
+				r.Logger.Error().
+					Err(err).
+					Msg("Unable to get nodes.")
+				nodeIds = make([]collector.Id, 0)
+			}
+			info.Ids = nodeIds
+			p.Nodes = &info
+
+			if len(nodeIds) > 0 {
```

Comment:
Original comment still useful here since this is confusing
```
// Since the serial number is bogus in c-mode
// use the first node's serial number instead (the nodes were ordered in getNodeUuids())
```